addons/godot-next/references/inflector.gd

```diff
@@ -0,0 +1,269 @@
+# author: xdgamestudios (adapted from C# .NET Humanizer, licensed under MIT)
+# license: MIT
+# description: Provides inflection tools to pluralize and singularize strings.
+# todo: Add more functionality
+# usage:
+# - Creating Inflector:
+#     inflector = Inflector.new() # new inflector with empty vocabulary
+#     inflector = Inflector.new(vocabulary) # new inflector with vocabulary
+#     Note:
+#     - Inflector with no vocabulary will not be able to apply convertion out-of-the-box.
+# - Creating Vocabulary:
+#     vocabulary = Inflector.Vocabulary.new() # creates new empty vocabulary
+#     vocabulary = Inflector.Vocabulary.build_default_vocabulary()
+#     Note:
+#     - Empty vocabulary can be manually configured with custom rules.
+#     - Default vocabulary will apply convertions for english language.
+# - Get Vocabulary:
+#     vocabulary = inflector.get_vocabulary() # reference to the vocabulary used by the inflector.
+# - Add Rules:
+#     vocabulary.add_plural(rule, replacement) # adds convertion rule to plural
+#     vocabulary.add_singular(rule, replacement) # adds convertion rule to singular
+#     vocabulary.add_irregular(rule, replacement) # adds irregular convertion
+#     vocabulary.add_uncountable(word) # add unconvertable word
+#     Note:
+#     - 'rule' is a String with regex syntax.
+#     - 'replacement' is a String with regex systax.
+# - Using Inflector:
+#     inflector.pluralize(word, singular = true) # returns the plural of the word
+#     inflector.singularize(word, plural = true) # returns the singular of the word
+#     Note:
+#     - The extra argument is used if there is no ensurance the given word is a singular or plural.
+extends Reference
+class_name Inflector
+
+##### CLASSES #####
+
+class Rule:
+	
+	##### PROPERTIES #####
+	
+	var _regex: RegEx
+	var _replacement: String
+	
+	##### NOTIFICATIONS #####
+	
+	func _init(p_rule: String, p_replacement: String):
+		_regex = RegEx.new()
+		#warning-ignore:return_value_discarded
+		_regex.compile(p_rule)
+		_replacement = p_replacement
+	
+	##### PUBLIC METHODS #####
+	
+	func apply(p_word: String):
```

Comment:
Return type hint.

Reply:
this can be string or null so it's variant!